code/game/objects/items.dm

```diff
@@ -643,98 +647,112 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
-		if(P.damage_type != STAMINA)// disablers dont do shit to shields
-			attackforce = (P.damage)
+		if(P.damage_type == STAMINA)
+			attackforce = 0 //Blocking disablers and tasers is free, but other projectiles do their standard damage
+
+	//Alright, it isn't a projectile, are we being hit with a weapon?
 	else if(isitem(hitby))
 		var/obj/item/I = hitby
-		attackforce = damage
-		if(I.is_sharp())
-			attackforce = (attackforce / 2)//sharp weapons get much of their force by virtue of being sharp, not physical power
-		if(!I.damtype == BRUTE)
-			attackforce = (attackforce / 2)//as above, burning weapons, or weapons that deal other damage type probably dont get force from physical power
-		attackforce = (attackforce * I.attack_weight)
-		if(I.damtype == STAMINA)//pure stamina damage wont affect blocks
+
+		//If we block a stamina weapon, it does nothing unless it electrocutes us separately
+		if(I.damtype == STAMINA)
 			attackforce = 0
-	else if(attack_type == UNARMED_ATTACK && isliving(hitby))
-		var/mob/living/L = hitby
-		if(block_flags & BLOCKING_NASTY && !HAS_TRAIT(L, TRAIT_PIERCEIMMUNE))
-			INVOKE_ASYNC(L, TYPE_PROC_REF(/atom, attackby), src, owner)
-			owner.visible_message(span_danger("[L] injures themselves on [owner]'s [src]!"))
-	else if(isliving(hitby))
-		var/mob/living/L = hitby
-		attackforce = (damage * 2)//simplemobs have an advantage here because of how much these blocking mechanics put them at a disadvantage
-		if(block_flags & BLOCKING_NASTY)
-			if(istype(L, /mob/living/simple_animal))
-				var/mob/living/simple_animal/S = L
-				if(!S.hardattacks)
-					INVOKE_ASYNC(S, TYPE_PROC_REF(/atom, attackby), src, owner)
-					owner.visible_message(span_danger("[S] injures themselves on [owner]'s [src]!"))
-			else
-				INVOKE_ASYNC(L, TYPE_PROC_REF(/atom, attackby), src, owner)
-				owner.visible_message(span_danger("[L] injures themselves on [owner]'s [src]!"))
-	owner.apply_damage(attackforce, STAMINA, blockhand, block_power)
-	if((owner.getStaminaLoss() >= 35 && HAS_TRAIT(src, TRAIT_NODROP)) || (HAS_TRAIT(owner, TRAIT_NOLIMBDISABLE) && owner.getStaminaLoss() >= 30))//if you don't drop the item, you can't block for a few seconds
-		INVOKE_ASYNC(owner, TYPE_PROC_REF(/mob/living/carbon/human, blockbreak))
-	if(attackforce)
-		owner.changeNext_move(CLICK_CD_MELEE)
+
+		//Blocking gets a bonus against weapons that don't get their power from brute force, but the weight also doesn't matter
+		else if(!I.damtype == BRUTE)
+			attackforce = (attackforce * 0.8)
+
+		//When blocking a sharp weapon, the force conveyed is determined purely by its weight rather than its damage
+		else if(I.is_sharp())
+			attackforce = I.w_class * 2
+
+		//And if it's a blunt weapon, blocking takes the worst outcome between weight and damage bonuses
+		else
+			attackforce = max(I.w_class * 3, attackforce * 1.5)
+
+		//Is it a weapon especially adept at counterattacks? If so we roll for one
+		if(block_flags & BLOCKING_COUNTERATTACK && prob(50))
```

Comment:
Add proper brackets to bitflag statements as they are dangerous
```suggestion
		if((block_flags & BLOCKING_COUNTERATTACK) && prob(50))
```

Reply:
Went through and contained all of them